server/datastore/mysql/hosts.go

```diff
@@ -4506,3 +4506,44 @@ func (ds *Datastore) GetMatchingHostSerials(ctx context.Context, serials []strin
 
 	return result, nil
 }
+
+func (ds *Datastore) GetHostHealth(ctx context.Context, id uint) (*fleet.HostHealth, error) {
+	sqlStmt := `
+		SELECT h.os_version, h.updated_at, h.platform, h.team_id, hd.encrypted as disk_encryption_enabled FROM hosts h
```

Comment:
You will need to select `h.team_id` too here, so that you fill the `TeamID` field for proper authorization.

Reply:
Hmm I think I am selecting `h.team_id`? Could this have been meant for a different line?

Reply:
Oh sorry, I was looking at it in the same order as the struct and somehow thought it would've been last and was missing. It's there allright! :D